File: app/src/main/java/com/example/javademogithubpractice/util/AppUtils.java


package com.example.javademogithubpractice.util;

import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;
import android.widget.Toast;

import androidx.annotation.NonNull;

import com.example.javademogithubpractice.AppApplication;
import com.example.javademogithubpractice.R;

import java.util.Locale;


public class AppUtils {
    @NonNull
    public static Locale getLocale(String language) {
        Locale locale;
        String[] array = language.split("-");
        if (array.length > 1) {
            //zh-rCN, zh-rTW", pt-rPT, etc... remove the 'r'
            String country =  array[1].replaceFirst("r", "");
            locale = new Locale(array[0], country);
        } else {
            locale = new Locale(language);
        }
        return locale;
    }

    public static void copyToClipboard(@NonNull Context context, @NonNull String uri) {
        ClipboardManager clipboard = (ClipboardManager) context.getSystemService(Context.CLIPBOARD_SERVICE);
        ClipData clip = ClipData.newPlainText(context.getString(R.string.app_github_name), uri);
        clipboard.setPrimaryClip(clip);
        Toast.makeText((AppApplication.get()), context.getString(R.string.success_copied),Toast.LENGTH_SHORT).show();
    }
}

File: KotlinDemoGithubProject/app/src/main/java/com/example/kotlindemogithubproject/util/AppUtils.kt
package com.example.kotlindemogithubproject.util

import android.content.*
import android.content.pm.PackageInfo
import android.content.pm.PackageManager
import android.content.res.Configuration
import android.content.res.Resources
import android.net.Uri
import android.os.Build
import android.view.View
import android.view.inputmethod.InputMethodManager
import androidx.annotation.NonNull
import androidx.annotation.Nullable
import com.example.kotlindemogithubproject.AppApplication
import com.example.kotlindemogithubproject.R

import java.util.Locale


object AppUtils {

    @NonNull
    fun getLocale(language: String): Locale {
        val locale: Locale
        val array = language.split("-".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        if (array.size > 1) {
            //zh-rCN, zh-rTW", pt-rPT, etc... remove the 'r'
            val country = array[1].replaceFirst("r".toRegex(), "")
            locale = Locale(array[0], country)
        } else {
            locale = Locale(language)
        }
        return locale
    }

    fun copyToClipboard(@NonNull context: Context, @NonNull uri: String) {
        val clipboard = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
        val clip = ClipData.newPlainText(context.getString(R.string.app_github_name), uri)
        clipboard.primaryClip = clip
    }



}
